src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -1189,4 +1205,42 @@ public String toString() {
     public LibraryTabContainer getLibraryTabContainer() {
         return tabContainer;
     }
+
+    /**
+     * Starts periodic monitoring of Git status for the database file
+     */
+    private void startGitStatusMonitoring() {
+        if (!bibDatabaseContext.isUnderVersionControl() || bibDatabaseContext.getDatabasePath().isEmpty()) {
+            return;
+        }
+
+        try {
+            // Create a unified status check task
+            Runnable checkGitStatusAndUpdateUI = () -> {
+                try {
+                    // Force refresh of Git status cache
+                    bibDatabaseContext.getGitStatus();
+
+                    // Update UI on JavaFX thread
+                    Platform.runLater(() -> updateTabTitle(changedProperty.getValue()));
+                } catch (Exception e) {
+                    // Silent fail
+                }
+            };
+
+            // Run initial check
+            checkGitStatusAndUpdateUI.run();
+
+            // Create the scheduled task for periodic updates
+            PauseTransition gitStatusTimer = new PauseTransition(Duration.seconds(10));
+            gitStatusTimer.setOnFinished(event -> {
+                checkGitStatusAndUpdateUI.run();
+                gitStatusTimer.play(); // Restart the timer
+            });
+            // Start periodic monitoring
+            gitStatusTimer.play();
+        } catch (Exception e) {
+            // Silent fail
```

Comment:
Classic "ignore all problems and hope for the best" approach? Silently discarding all exceptions - because debugging is overrated.
https://github.com/HugoMatilla/Effective-JAVA-Summary?tab=readme-ov-file#65-dont-ignore-exceptions

Do not fail silently.